volmdlr/edges.py

```diff
@@ -1621,26 +1624,48 @@ def point_belongs(self, point2d, abs_tol=1e-10):
         """
-        vector_start = self.start - self.center
-        vector_end = self.end - self.center
-        vector_point = point2d - self.center
-        r1 = vector_start.norm()
-        cp = vector_point.norm()
-        if math.isclose(cp, r1, abs_tol=abs_tol):
-            if (self.start.x < self.end.x) and (self.start.y < self.end.y):
-                arc_angle = - volmdlr.core.clockwise_angle(vector_start,
-                                                           vector_end)
-                point_angle = - volmdlr.core.clockwise_angle(vector_start,
+
+        if (self.start.point_distance(point2d) <= abs_tol
+                or self.end.point_distance(point2d) <= abs_tol):
+            return True
+
+        if self.is_trigo:
+            arc_trigo = self.reverse()
+        else:
+            arc_trigo = self
+
+        vector_start = arc_trigo.start - arc_trigo.center
+        vector_end = arc_trigo.end - arc_trigo.center
+        vector_point = point2d - arc_trigo.center
+        radius = vector_start.norm()
+        center_point_dis = vector_point.norm()
+
+        if math.isclose(center_point_dis, radius, abs_tol=abs_tol):
+            arc_angle = volmdlr.core.clockwise_angle(vector_start,
+                                                     vector_end)
+            point_start_angle = volmdlr.core.clockwise_angle(vector_start,
                                                              vector_point)
-            else:
-                arc_angle = volmdlr.core.clockwise_angle(vector_start,
-                                                         vector_end)
-                point_angle = volmdlr.core.clockwise_angle(vector_start,
-                                                           vector_point)
-            if point_angle <= arc_angle:
+            point_end_angle = volmdlr.core.clockwise_angle(vector_point,
+                                                           vector_end)
+            if math.isclose(arc_angle, point_start_angle + point_end_angle, abs_tol=abs_tol):
                 return True
         return False
 
+        # if math.isclose(center_point_dis, radius, abs_tol=abs_tol):
+        #     if (arc_trigo.start.x < arc_trigo.end.x) and (arc_trigo.start.y < arc_trigo.end.y):
+        #         arc_angle = - volmdlr.core.clockwise_angle(vector_start,
+        #                                                    vector_end)
+        #         point_angle = - volmdlr.core.clockwise_angle(vector_start,
+        #                                                      vector_point)
+        #     else:
+        #         arc_angle = volmdlr.core.clockwise_angle(vector_start,
+        #                                                  vector_end)
+        #         point_angle = volmdlr.core.clockwise_angle(vector_start,
+        #                                                    vector_point)
+        #     if point_angle <= arc_angle:
+        #         return True
+        # return False
+
```

Comment:
I believe you can delete this commented block

Reply:
Done !